decbinhex4/addons/main/src/code/convert.nim

```diff
@@ -2,28 +2,24 @@ import math, strutils
 import std/times #benchmark only
 
 # the asterisk is so it can be called from another file when imported
-proc convert_from_decimal*(input_decimal: string): string =
-  var result_binary: string = "reset binary result"
-  var result_hex: string = "reset hex result"
-  if input_decimal.len < 20:
-    result_binary = strip((input_decimal.parseInt).toBin(63), trailing = false, chars = {'0'})
+proc convert_from_decimal*(input_decimal: string):string =
+  var result_binary: string = "reset binary result"; var result_hex: string = "reset hex result"
+  if input_decimal.len < 20: result_binary = strip((input_decimal.parseInt).toBin(63), trailing = false, chars = {'0'})
   else: result_binary = "Output binary too large!"
   result_hex = strip(input_decimal.parseInt.toHex, trailing = false, chars = {'0'})
   return("Binary is: " & result_binary & "\nHexadecimal is: " & result_hex)
 
-proc convert_from_binary*(input_binary: string): string =
-  var result_decimal: int = 0
-  var result_hex: string = "reset hex result"
+proc convert_from_binary*(input_binary: string):string =
+  var result_decimal: int = 0; var result_hex:string = "reset hex result"
   for i, c in input_binary:
     #use length to invert i... to get place without a decreasing variable
     if c == '1': result_decimal += 2^abs(i-(input_binary.len-1))
   result_hex = strip(result_decimal.toHex, trailing = false, chars = {'0'})
   return("Decimal is: " & $result_decimal & "\nHexadecimal is: " & result_hex)
 
-proc convert_from_hexadecimal*(input_hex: string): string =
-  var lengthvar_hex: int = input_hex.len-1
-  var result_binary: string = "reset binary result"
-  var result_decimal: int = 0
+proc convert_from_hexadecimal*(input_hex:string):string =
+  var result_binary:string = "reset binary result"; var result_decimal:int = 0
+  var lengthvar_hex:int = input_hex.len-1
```

Comment:
Use this section to illustrate the proposal for variable definitions.
I think this coding style will also keep the semantic cohesion:

```nim
var
  result_binary: string = "reset binary result"
  result_decimal: int = 0
var
  lengthvar_hex: int = input_hex.len-1
```

Reply:
The combined line is doing the same thing, putting further emphasis on the line that isn't combined.

However, I believe it may be clearer if I don't assign values:

`var result_decimal: int; var result_hex:string`

Reply:
If you really do not want multiple lines, try the following:

```nim
var (result_binary, result_decimal): (string, int) = ("reset binary result", 0)
var lengthvar_hex: int = input_hex.len-1
```

However, maintenance is still a hassle, so it is not recommended.

Reply:
I just changed the line that used the same types to `  var result_binary, result_hex:string`.

I already changed the other 2 lines in that file to only initialize.